addons/analytic/models/analytic_plan.py

```diff
@@ -174,6 +174,13 @@ def _compute_children_count(self):
         for plan in self:
             plan.children_count = len(plan.children_ids)
 
+    @api.onchange('parent_id')
+    def _onchange_parent_id(self):
+        project_plan, __ = self._get_all_plans()
+        for plan in self:
```

Comment:
IIRC, `onchange` runs for 1 record at a time, so we don't need to loop over self.